WasatchNET/MultiChannelWrapper.cs

```diff
@@ -0,0 +1,407 @@
+                positions.Add(pos);
+            }
+
+            // default to internal triggering
+            triggeringEnabled = false;
+
+            return spectrometerCount > 0;
+        }
+
+        /// <summary>
+        /// Release all resources at end of application session.
+        /// </summary>
+        public void close()
+        {
+            driver.closeAllSpectrometers();
+            reset();
+        }
+
+        ////////////////////////////////////////////////////////////////////////
+        // Device Control
+        ////////////////////////////////////////////////////////////////////////
+
+        /// <summary>
+        /// How many spectrometers (channels) were found.
+        /// </summary>
+        public int spectrometerCount => specByPos.Count;
+
+        // private utility method to sort spectrometers by integration time
+        // in increasing order, so we can process the "soonest-done" first,
+        // grab the longest at the end and save time.
+        IEnumerable<Spectrometer> specByIntegrationTime()
+        {
+            return from pair in specByPos
+                   orderby pair.Value.integrationTimeMS ascending
+                   select pair.Value;
+        }
+
+        /// <summary>
+        /// Lets the caller directly control the spectrometer at a given position.
+        /// </summary>
+        /// <param name="pos">Which channel</param>
+        public Spectrometer getSpectrometer(int pos)
+        {
+            return specByPos.ContainsKey(pos) ? specByPos[pos] : null;
+        }
+
+        /// <summary>
+        /// The position of the spectrometer configured to generate the external hardware trigger.
+        /// </summary>
+        /// <returns>-1 if none found</returns>
+        public int triggerPos => specTrigger != null ? specTrigger.multiChannelPosition : -1;
+
+        /// <summary>
+        /// Whether external hardware triggering is currently enabled.
+        /// </summary>
+        public bool triggeringEnabled
+        {
+            get => _triggeringEnabled; 
+            set
+            {
+                foreach (var pair in specByPos)
+                {
+                    logger.debug($"pos {pair.Key}: triggerEnabled -> {value}");
+                    pair.Value.triggerSource = value ? TRIGGER_SOURCE.EXTERNAL : TRIGGER_SOURCE.INTERNAL;
+                }
+                _triggeringEnabled = value;
+            }
+        }
+        bool _triggeringEnabled;
+
+        /// <summary>
+        /// The position of the spectrometer configured to control the fans.
+        /// </summary>
+        /// <returns>-1 if none found</returns>
+        public int fanPos => specFan != null ? specFan.multiChannelPosition : -1;
+
+        /// <summary>
+        /// Use the configured spectrometer's laserEnable output to turn the system fans on or off.
+        /// </summary>
+        /// <remarks>Obviously dangerous on spectrometers with physical lasers.</remarks>

```

Comment:
I do not like this, but if it's how we're getting it done right now that's cool

Reply:
I do not like it, Sam-I-Am!  I do not like our Technical Debt. Not here or there, it makes me fret.